src/main/interface/msp.c

```diff
@@ -1795,6 +1795,7 @@ static mspResult_e mspProcessInCommand(uint8_t cmdMSP, sbuf_t *src)
                             vtxCommonSetPitMode(vtxDevice, newPitmode);
                         }
                     }
+                    saveConfigAndNotify();
```

Comment:
This should not save here. The correct fix for this is to change the lua scripts to save this when 'Save' is selected.

Reply:
Care to elaborate? Have you read the Lua code? I have and the LUA script doesn't do any fancy saving when changing pids or rates. It just uses the MSP_SET_PID_ADVANCED.

Reply:
Yes, and then you select 'Save' in the menu that pops up when you long press 'Menu'.

Reply:
Wooooow, you really misunderstand the problem. The fact that you have to save in the Lua script is obvious and that is not the problem. The problem is that the the FC doesn't "remmember" the setting for the next time it boots up. And sets the old channel it had to the vtx.

You see when using the OSD the problem isn't there because the user uses "save and exit" that saves the vtx settings.

If you look at spektrumVtxControl() wich is a a different function for setting vtx settings you will notice the "saveConfigAndNotify();" at the end. That thing is missing in the MSP protocol. The Lua script uses the MSP protocol, so we also need to save this setting.

Reply:
I did understand the problem very well. Just breaking how MSP works by forcing it to save in a command that should not save is not the solution. Closing this.